[FIX] Clarify MEG empty-room recommendations (#1125)
bids-standard/bids-specification: src/04-modality-specific-files/02-magnetoencephalography.md

## Empty-room MEG recordings

Empty-room MEG recordings capture the environmental and recording system's
noise.

> In the context of BIDS it is RECOMMENDED to perform an empty-room recording for

Reviewer: A single empty room recording should exist for any given session. Not more.

Reviewer 2: I disagree with this.  Are you saying that if I collect multiple noise recordings, I can't have them all in my BIDS dataset?  See also my comment below about the AssociatedNoiseRecording field.

> In the context of BIDS it is RECOMMENDED to perform an empty-room recording for each experimental session.

Reviewer 3: Do really recommend to acquire noise recordings before each **session**, i.e. a subject does 3 runs x 3 tasks, you record 9 noise sessions? Or does "experimental session" refer to "all the contiguous sessions of a subject"?
Is this how you do it at the Donders?

I had the impression that at many sites, this is more of a daily thing, done by the lab manager also as a way to track the technical health of the MEG system.

Reviewer 4: yes, would be important to be precise here also with regards to what ["session" means in BIDS](https://bids-specification.readthedocs.io/en/latest/02-common-principles.html#definitions)

Reviewer 5: > Do really recommend to acquire noise recordings before each **session**, i.e. a subject does 3 runs x 3 tasks, you record 9 noise sessions?

My understanding is that one "session" is one visit to the lab

Reviewer 3: > My understanding is that one "session" is one visit to the lab

Yes, obviously. I got distracted... Please ignore this previous comment about the session.

> In the context of BIDS it is RECOMMENDED to perform an empty-room recording for each experimental session.

Author: There are two recommendations, one on the lab practice and one on the data organization, and the two are not really aligned. These came out of [BEP008](https://bids.neuroimaging.io/bep008), see page 15, which happened to be the first big BEP that was merged. It might have been better had it been captured in 2018 as not being optimal.

The two practices are both being used, and I personally recommend that everyone does in their lab whatever works for them. I think it would be better if BIDS would restrict itself to data organization, and not to lab procedures. But simply scrapping the "empty-room recording per session" from the spec might lead to the impression that that is not recommended, whereas it actually was the recommended strategy (by consensus) in BEP006.

Should we rephrase, say that both are possible, that "empty-room recording per session" is recommended (to stay in line with BEP006) and give an example for both? Actually, compared to the current text in this PR the only change would then be to also give an example for the "empty-room recording per session".

> In the context of BIDS it is RECOMMENDED to perform an empty-room recording for each experimental session.

Reviewer 2: I agree with @robertoostenveld.  Still, how would you feel about changing a single word:  "In the context of BIDS it is RECOMMENDED to **associate** an empty-room recording for each experimental session."  This implicitly maintains the idea that it should be performed, while bringing it back to the organization context of BIDS, and the statement now works for both practices.

If empty-room recordings are not collected for each individual experimental session, one empty-room recording may be used with multiple subjects and/or sessions.

> For example, they may be collected once a day.

Reviewer: In case of conflict (a sub-emptyroom exists, and a noise recording exist inside the session of interest), the noise recording inside the session folder should supersede the sub-emptyroom.

> In that case it is RECOMMENDED to store the empty-room recording inside a subject directory named `sub-emptyroom`.

Reviewer 3: "In that case": does that imply that in the former case, it is RECOMMENDED to store it in the subject folder together with the other tasks? If this is the case, maybe it should be made more explicit?

Would it make sense to separate the two typical cases even more clearly, with headers: 
 - case 1: one noise recordings per session folder
 - example 1
 - case 2: one noise recordings per day
 - example 2
?

Reviewer 5: > Would it make sense to separate the two typical cases even more clearly, with headers?

I like this idea.

Reviewer: An example 3 could illustrate the case when there is a conflict between a local noise recording in the session, and a sub-emptyroom. Automatically assigning the correct noise recording to a given task recording needs arbitration.

A policy like "if there is no noise recording in a session, and several noise recordings are present in sub-emptyroom for a given day, then the noise recording closest in time should be assigned to the session's recordings".

Author: BIDS uses the 80/20 Pareto Principle: we don't have to deal with all possible situations that might arise (as then we'll continue discussing indefinitely), but only with those 20% of the cases that result in 80% of the value. I think your case 3 falls outside the 80% of the value.

Author: thanks for the suggestion for splitting up the examples @dnacombo. With commit 0aa328e85799aa5f0a951545c8838fddc7ee1eba I have made the two cases more explicit in the text as well.

The label for the [`task-<label>`](../99-appendices/09-entities.md#task) entity in the empty-room recording SHOULD be set to `noise`.
If a [`session-<label>`](../99-appendices/09-entities.md#ses) entity is present, its label SHOULD be the date of the empty-room recording in the format `YYYYMMDD`, that is `ses-YYYYMMDD`.
The `scans.tsv` file containing the date and time of the acquisition SHOULD also be included.
The rationale is that this naming scheme will allow users to easily retrieve the empty-room recording that best matches a particular experimental session, based on date and time of the recording.
It should be possible to query empty-room recordings just like usual subject recordings, hence all metadata sidecar files (such as the `channels.tsv`) file SHOULD be present as well.

> Example 1:

Reviewer: One noise recording per day applying to all subjects for that day.

Author: this has been addressed in 0aa328e85799aa5f0a951545c8838fddc7ee1eba

## Empty-room MEG recordings

Empty-room MEG recordings capture the environmental and recording system's noise.

> In the context of BIDS it is RECOMMENDED to associate an empty-room recording to each experimental session (see `AssociatedEmptyRoom` field in the `*_meg.json` sidecar file).

Reviewer 5: I rewrote this. But I'm not sure how to produce the correct link. Could sb please help?

```suggestion

It is RECOMMENDED to explicitly specify which empty-room recording should be used with which experimental run(s) or session(s). This can be specified via the [`AssociatedEmptyRoom`](99-appendices/14-glossary.html#associatedemptyroom-metadata) field in the `*_meg.json` sidecar files.
```

Reviewer 4: You had it almost right, we just need to:

- add a "SPEC_ROOT/" in front of internal links
- use `.md` as extensions, instead of html

Reviewer 4: ah no ... "SPEC_ROOT" was wrong, ... `../` was needed instead. We only need "SPEC_ROOT" when using internal links from within schema files.

> Empty-room recordings may be collected once per day, where one empty-room recording may be used with multiple subjects and/or sessions (see example 1).

Reviewer 5: ```suggestion
Empty-room recordings may be collected once per day, where a single empty-room recording may be shared between multiple subjects and/or sessions (see example 1).
```

If a [`session-<label>`](../99-appendices/09-entities.md#ses) entity is present, its label SHOULD be the date of the empty-room recording in the format `YYYYMMDD`, that is `ses-YYYYMMDD`.
The `scans.tsv` file containing the date and time of the acquisition SHOULD also be included.
The rationale is that this naming scheme will allow users to easily retrieve the empty-room recording that best matches a particular experimental session, based on date and time of the recording.
It should be possible to query empty-room recordings just like usual subject recordings, hence all metadata sidecar files (such as the `channels.tsv`) file SHOULD be present as well.

In the case of empty-room recordings being collected for the individual experimental session, it is recommended to store the empty-room recording along with that subject and session.

In either case, the label for the [`task-<label>`](../99-appendices/09-entities.md#task) entity in the empty-room recording SHOULD be set to `noise`.

Example 1:

> One noise recording per day, applying to all subjects for that day.

Reviewer 5: ```suggestion
One empty-room recording per day, applying to all subjects for that day.
```

noise.
In the context of BIDS it is RECOMMENDED to perform an empty-room recording for
each experimental session.

> If empty-room recordings are not done prior to each individual experimental session, but

Reviewer 2: "If empty-room recordings are not collected for each individual experimental session, one empty-room recording may be used with multiple subjects and/or sessions. For example, they may be collected once a day."

Author: thanks, I used your text.

on a daily schedule, one empty-room recording may have to be used with multiple subjects and/or sessions. In that case

> it is RECOMMENDED to store the empty-room recording inside a subject directory

Reviewer 5: Store it inside … which subject's directory? The `sub-emptyroom`? or `sub-participant`? I find this now confusing in combination with the two examples provided.

Reviewer 5: Ooooh!! I actually missed these last two words! Sorry about the confusion!!

   }
) }}

> Example 2:

Reviewer: One recording per session, stored within the session folder.

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->

> {{ MACROS___make_filetree_example(

Reviewer 6: My suggestion is just to add a note here that the empty room recordings can be symlinked and note that it might be helpful in the case where two empty room recordings correspond to the same subject to save data space

Reviewer 6: Not everyone might know who is reading this spec without looking it up that symlinks are allowed in BIDS

Reviewer 5: @alexrockhill I'd be against adding this here, this is really an "implementation detail" that won't even be platform-agnostic and portable. If a Linux user zips a BIDS structure containing symlinks and sends it to their Windows friends, they'll often into issues (depending on how the zip file was created)

Let's open a separate discussion on the very issue of symlinks, no? :)

Reviewer 6: Sounds good

Author: are symbolic links allowed? I searched the specification, the word "symlink" or "symbolic" is nowhere to be found. Symbolic links would indeed not work on all file systems and operating systems. Great if they work for your use-case, but  I am not sure whether they should be included in the specification. 

Please do start a separate issue so that non-MEG folks can chime in as well. It would not surprise me if this has already been discussed in the past and that someone can share insights from the past.

   }
) }}

Example 2:

> 

Reviewer 7: Sorry I did my suggestions as a commit, instead of as a review.
